urllib3/_async/connection.py

```diff
@@ -311,8 +311,7 @@ def __init__(self, host, port, backend=None,
                  source_address=None, tunnel_host=None, tunnel_port=None,
                  tunnel_headers=None):
         self.is_verified = False
-
-        self._backend = backend or SyncBackend()
+        self._backend = backend or load_backend("sync")
```

Comment:
Question: if this function constantly calls `from .sync_backend import SyncBackend` - is it constantly re-running the import machinery whenever we create a connection?

If so, we should special case `SyncBackend` to skip the import machinery in the interests of making the common case more efficient.

Alternatively, we should ensure that users have always specified a backend by this point (which is how I plan for async backends to work), and just leave the `or` case for unit tests.

Reply:
Re-importing an already-imported module is pretty cheap, because python immediately notices that it already has the loaded module cached and uses that. So it's like a few dict lookups.

Reply:
Ok, also: what we actually want the logic to be here (I think) is:

* If this is the sync version of the code, then the backend must be `None` or `"sync"` or `Backend("sync")`. Anything else is an error.

* If this is the async version of the code, then the backend must be specified, and must be *not* the sync backend.

And... we kind of want to do this check inside all of the classes that take `backend` arguments.

So I suggest adding a `normalize_backend` function that takes a user `backend` argument, and returns a valid `Backend` or else an error.

The trickiest bit is detecting whether we are in sync mode or async mode. Eventually we should add a nice way to do this to the code generation script, but for now one can do:

```python
async def f():
    return None

obj = f()
if obj is None:
    SYNC_MODE = True
else:
    # prevent unawaited coroutine warning
    obj.close()
    SYNC_MODE = False
```